src/http/http_endpoint.h

```diff
@@ -74,8 +85,10 @@ namespace http
         }
         catch (const std::exception& e)
         {
-          LOG_FAIL_FMT("Error parsing HTTP request");
-          LOG_DEBUG_FMT("Error parsing HTTP request: {}", e.what());
+          if (error_reporter)
+          {
+            error_reporter->report_parsing_error(session_id);
+          }
```

Comment:
```suggestion
          }
          LOG_DEBUG_FMT("Error parsing HTTP request: {}", e.what());
```

We should keep the debug log with the precise error message.